### PR TITLE
closes #89 lasta@devel

### LINKED ISSUE #89: Questioning the need for derive_param_lasta() from tester feedback
Stefan: At Roche we would need to use the source_pd argument for the derivation of the LSTAC parameter. However, this parameter is not used anywhere. So I would question it. **(needs to be checked with Roche data standards to understand why we have this in our global standards)**
Furthermore, I would question if we need derive_param_lasta() at all. The Roche testers requested an update of derive_param_first_event() (https://github.com/pharmaverse/admiral/issues/1317) which covers the derive_param_lasta() functionality (except the source_pd part).

### PR DESCRIPTION
**Pull request by rossfarrugia**
Please add "Closes #<issue number>" to the title of the pull request. Then the
issue is closed automatically once it is merged to `main`.

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiralonco/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

The following line requests to update the man pages by the "Man Pages" workflow.

/roxygenize

### CODE REVIEW

**Comment by rossfarrugia** [comment]
@bundfussr i'm more inclined to go with issue an error in this case, as the package is so early anyway without many users likely - agree?

**Comment by rossfarrugia** [comment]
@bundfussr i'm happy with these updates so feel free to take a look but will have to wait for the CI workflows / gh actions to be back working again

**Comment by rossfarrugia** [comment]
thanks @bundfussr - i was so focused on the deprecation step that i missed such obvious things in the templates/vignettes! will get on these updates tomorrow and hopefully the workflows will be back working then too

File: inst/templates/ad_adrs.R [hunk 1]
```diff
@@ -288,8 +288,11 @@ adrs <- adrs %>%
 
 # Last disease assessment
 adrs <- adrs %>%
-  derive_param_lasta(
+  derive_param_extreme_event(
+    dataset_adsl = adsl,
+    dataset_source = adrs,
     filter_source = PARAMCD == "OVR" & ANL01FL == "Y",
+    order = vars(ADT, RSSEQ),
```
[the remark this answers is not in the document]
Comment (rossfarrugia) [reply inst/templates/ad_adrs.R:295]: for the dummy bit please can you create a new issue for next release to update `derive_param_extreme_event()` accordingly and then we can come back to simplify this in the template. i'll need the same update in admiralroche template too - i missed this there as well

**Comment by rossfarrugia** [comment]
@bundfussr i spotted a couple of extra updates needed for DEATH parameter too, which i'll make for admiralroche as well - see the last 2 commits. just need to check workflows now and then will request a re-review once they run ok

**Comment by rossfarrugia** [comment]
@bundfussr is the R 4.0 CI fail a known issue? looks to be something to do with staged.dependencies as its not picking up admiral and admiraldev dependencies

**Comment by bundfussr** [comment]
@rossfarrugia , yes, that's a known issue. Dinakar is investigating it.

**Comment by rossfarrugia** [comment]
@bundfussr i just spotted that the update i made for DEATH parameters to set `AVAL` is needed for all the places where we use `derive_param_extreme_event()` - what do you think about me doing this just once at the end for all params with Y or N value as a catch all instead? ideally the function itself would give `AVAL` as well as `AVALC` in a future release.

**Comment by bundfussr** [comment]
@rossfarrugia , fine with me.

I think the function should support both deriving `AVAL` and not deriving `AVAL`. We should consider this in the clean-up issue.

**Comment by rossfarrugia** [comment]
@bundfussr should be ok to re-review now.

File: inst/templates/ad_adrs.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (bundfussr) [reply inst/templates/ad_adrs.R:295]: Issue created: https://github.com/pharmaverse/admiral/issues/1630

File: vignettes/adrs.Rmd [hunk 2]
```diff
@@ -788,6 +794,30 @@ dataset_vignette(
...
+      TRUE ~ AVAL
+    )
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = vars(USUBJID, PARAMCD, AVALC, AVAL),
+  filter = USUBJID == "01-701-1015" & AVALC %in% c("Y", "N")
```
Comment (bundfussr) [inline vignettes/adrs.Rmd:817]: Should we display all parameters to demonstrate that `AVAL` for non-binary parameter is not changed?

**Review by bundfussr — CHANGES_REQUESTED** [review]
Could you copy the deprecation badges to `man/figures`? You can take them from admiral. This should fix the broken image on the `derive_param_lasta()` page.

File: vignettes/adrs.Rmd (hunk 2 above)
Comment (rossfarrugia) [reply vignettes/adrs.Rmd:817]: no i think it's ok and users would assume that i'm sure.

**Pull request merged**